cartconditions/Coupon.php

```diff
@@ -92,12 +93,28 @@ public function beforeApply()
 
     public function getActions()
     {
-        return [
-            [
-                'value' => optional($this->getModel())->discountWithOperand(),
-                'calculateValue' => [$this, 'calculateValue'],
-            ],
+        $value = optional($this->getModel())->discountWithOperand();
+
+        // if we are item limited and not a % we need to apportion
+        if (stripos($value, '%') === false AND optional($this->getModel())->is_limited_to_cart_item) {
+
+            $applicableItems = self::$applicableItems;
+            $applicableItemsTotal = Cart::content()->sum(function ($cartItem) use ($applicableItems) {
+                if (!$applicableItems->contains($cartItem->id))
+                    return 0;
+
+                return $cartItem->subtotalWithoutConditions();
+            });
+
+            $value = ($this->target->subtotalWithoutConditions() / $applicableItemsTotal) * (float)$value;
+        }
```

Comment:
Works nicely! 

Move this logic into a separate method. I don't think we need line 114, the `calculateValue` method is not defined anywhere.

Reply:
Just to add, this `if (!$applicableItems->contains($cartItem->id))` throws an error when `$applicableItems` is null.

Reply:
Should be changed